src/shared/logs-show.c

```diff
@@ -450,6 +450,9 @@ static void parse_display_realtime(
         assert(j);
         assert(ret);
 
+        // FIXME: _SOURCE_MONOTONIC_TIMESTAMP is in CLOCK_BOOTTIME, hence we cannot use it for adjusting realtime.
+        source_monotonic = NULL;
```

Comment:
hmm, i am not sure i grok this comment. i mean we shouldn't really fix this up when parsing, but simply fill in the correct fields.

note there has been a long standing issue that we probably should include the CLOCK_BOOTTIME timestamp in all journal entries too (right now we only store the realtime + monotonic timestamp). There's even a bug open about that somewhere I think, was unable to find it quickly though.

I couldn't come up with a nice way to include the boottime timestamp in the records without massively breaking the format and increasing the entry record size so far. The least bad idea i had, is that depending on some feature flag we'd use the upper 32bit of seqnum, plus the upper 8bit of the monotonic + realtime fields of EntryObject to encode the offset of boottime towards seqnum. That would give us 32+8+8 = 48bit accuracy, i.e we'd allow 9 years or so of delta between CLOCK_MONOTONIC and CLOCK_BOOTTIME. i.e. people could hibernate their system 9 year long before the journal couldn't deal with it anymore.

Reply:
Yeah, I am considering something similar now. Saving shift is slightly complicated, but can keep entry object size, hence maybe good idea.